src/block/icon-list/edit.js

```diff
@@ -158,34 +171,53 @@ const edit = props => {
 	const {
 		className,
 		setAttributes,
+		attributes,
 	} = props
 
 	const {
-		icon,
-		text,
 		design = '',
 		displayAsGrid = false,
 	} = props.attributes
 
 	const mainClasses = classnames( [
 		className,
 		'ugb-icon-list--v2',
-		`ugb-icon--icon-${ icon }`,
 	], applyFilters( 'stackable.icon-list.mainclasses', {
 		'ugb-icon-list--display-grid': displayAsGrid,
 	}, design, props ) )
 
-	return (
-		<BlockContainer.Edit className={ mainClasses } blockProps={ props } render={ () => (
-			<Fragment>
+	const ListItem = range( 1, 7 ).map( index => {
+		const icon = attributes[ `icon${ index }` ]
+		const text = attributes[ `text${ index }` ]
+		return (
+			<li
+				className={ `ugb-icon-list--item${ index }` }
+				key={ index }
+			>
+				<SvgIconPlaceholder
+					className={ `ugb-icon-list--icon${ index }` }
+					attrNameTemplate="icon%s"
+					blockAttributes={ props.attributes }
+					value={ icon }
+					onChange={ value => setAttributes( { [ `icon${ index }` ]: value } ) }
+				/>
 				<RichText
-					tagName="ul"
-					multiline="li"
-					value={ text }
-					onChange={ text => setAttributes( { text } ) }
+					className={ `ugb-icon-list--text${ index }` }
```

Comment:
Follow [BEM](http://getbem.com/introduction/)

```suggestion
					className={ `ugb-icon-list__text${ index }` }
```